oidc.go

```diff
@@ -101,8 +102,14 @@ func NewProvider(ctx context.Context, issuer string) (*Provider, error) {
 		return nil, fmt.Errorf("%s: %s", resp.Status, body)
 	}
 	defer resp.Body.Close()
+
 	var p providerJSON
 	if err := json.Unmarshal(body, &p); err != nil {
+		var isJSON bool
+		content, _, _ := mime.ParseMediaType(r.Header.Get("Content-Type"))
```

Comment:
Although, we might need to check the error case based on your comment below, in order to correctly provide the content-type in the error message. I'll update.